network/src/commonMain/kotlin/com/wire/kalium/network/utils/ObfuscateUtil.kt

```diff
@@ -0,0 +1,92 @@
+@file:Suppress("TooGenericExceptionCaught")
+package com.wire.kalium.network.utils
+
+import com.wire.kalium.logger.obfuscateId
+import com.wire.kalium.logger.obfuscateUrlPath
+import com.wire.kalium.network.kaliumLogger
+import io.ktor.http.Url
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonObject
+
+fun obfuscateAndLogMessage(text: String) {
+    try {
+        val obj = (Json.decodeFromString(text) as JsonElement)
+        if (obj.jsonArray.size > 0) {
+            obj.jsonArray.map {
+                logObfuscatedJsonElement(it)
+            }
+        } else {
+            logObfuscatedJsonElement(obj)
+        }
+    } catch (e: Exception) {
+        "error the body content while logging "
+    }
+}
+
+fun logObfuscatedJsonElement(obj: JsonElement) {
+    obj.jsonObject.entries.toMutableSet().map {
+        when {
+            sensitiveJsonKeys.contains(it.key.lowercase()) -> {
+                kaliumLogger.v("${it.key} : ******")
+            }
+
+            sensitiveJsonIdKeys.contains(it.key.lowercase()) -> {
+                kaliumLogger.v("${it.key} : ${it.value.toString().obfuscateId()}")
+            }
+
+            sensitiveJsonObjects.contains(it.key.lowercase()) -> {
```

Comment:
what if the key is uppercase?

Reply:
in this cases the user id is not UUID, same above